zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java

```diff
@@ -96,9 +96,7 @@ public static Path path(final URI uri) throws IOException {
    * <p>NOTE: Zoltar internal use only!</p>
    */
   public static URI downloadIfNonLocal(final URI path) throws IOException {
-    final String fixedPath =
-        path.toString().endsWith("/") ? path.toString() : path.toString() + "/";
```

Comment:
@rculbertson so trailing slash here is needed to support :

* `gs://buck_name/dir`
* `gs://buck_name/dir/`

Maybe we can just move this logic to `path` method and add the trailing slash based on the loaded provider?

Reply:
@regadas That's a good idea. How about adding the slash for URIs that have the scheme `gs`? Just push a commit with that change.

Reply:
@regadas Ah, my bad. I totally missed that. I was trying to avoid manually fixing paths, but I guess there's no avoiding it. I refactored the PR to handle that case.

BTW - thanks for adding the GCS integration test! Very helpful!

Reply:
yeah I knew there was a reason why we had the ugly thing.